models/dojo.py

```diff
@@ -0,0 +1,277 @@
+        room_type = arg["<room_type>"]
+        room_names = arg["<room_name>"]
+        for room_name in room_names:
+            if room_name == ('office' or 'livingspace'):
+                raise Exception("This is not a valid room name")
+            else:
+                if room_type.upper() == "OFFICE":
+                    self.office_rooms[room_name] = Office(room_name)
+
+                elif room_type.upper() == "LIVINGSPACE":
+                    self.living_rooms[room_name] = LivingSpace(room_name)
+
+                else:
+                    raise Exception("Invalid Room Type.Must be office or living")
+
+                Prefix = ("A" if room_type.upper() == "LIVINGSPACE" else "An")
+                print("{} {} called {} has been successfully created".format(
+                    Prefix, room_type, room_name))
+
+    def add_person(self, arg):
+        """The function adds a person to a room randomly."""
+        first_name = arg["<first_name>"]
+        last_name = arg["<last_name>"]
+        rank = (arg["<FELLOW/STAFF>"])
+        wants_accomodation = (arg["<wants_accomodation>"] or "N")
+
+        if rank.upper() == "STAFF":
+            new_user = Staff(first_name, last_name)
+            staff_id = "S" + str(new_user.id)
+            self.persons[staff_id] = new_user
+
+            print("{0} {1} has been successfully added".format(self.persons[
+                staff_id].rank, self.persons[staff_id]))
+
+        elif rank.upper() == "FELLOW":
+            new_user = Fellow(first_name, last_name,
+                              wants_accomodation.upper())
+            fellow_id = "F" + str(new_user.id)
+            self.persons[fellow_id] = new_user
+
+            print("{0} {1} has been successfully added".format(self.persons[
+                fellow_id].rank, self.persons[fellow_id]))
+
+        else:
+            raise Exception('Person can only be a fellow or staff')
+
+        self.assign_person(new_user)
+
+    @staticmethod
+    def get_available_room_spaces(room_spaces):
+        available_room_spaces = []
+        for room, room_info in room_spaces.items():
+            if (len(room_info.occupants) < room_info.max_occupants):
+                available_room_spaces.append(room_info)
+        return available_room_spaces
+
+    def assign_person(self, person):
+        """The function randomly assigns a person to a room."""
+        available_office_spaces = self.get_available_room_spaces(
+            self.office_rooms)
+        if available_office_spaces == []:
+            print("There is currently no office space")
+        else:
+            assigned_office_space = random.choice(available_office_spaces)
+            assigned_office_space.occupants.append(person)
+            person.office_space_allocated = assigned_office_space.room_name
+            print("{0} has been allocated the office {1}".format(
+                person.first_name, assigned_office_space.room_name))
+
```

Comment:
for the above if else block, it is a better practice to catch for positive outcomes and let your negative cases fall to the else. 
```
if available_office_spaces:
...
else
...
```